### ISSUE TITLE
Latest release of dynaconf breaks kedro

### ISSUE THREAD

**Issue by FlorianGD** [quantumblacklabs/kedro#888]
## Description
Dynaconf version 3.1.6 and 3.1.7 breaks the project validator from Kedro.

## Context
Any command using kedro is broken with those dependencies. In the `requirements.txt` from Kedro, the constraint is `dynaconf~=3.1.2`, so those versions should be allowed.

## Steps to Reproduce
In a fresh environment with kedro `0.17.4`
1. kedro new starter=pandas-iris
  I kept the default settings
2. cd new-kedro-project
3. pip install -U -r src/requirements.txt
4. pip show dynaconf

```console
Name: dynaconf
Version: 3.1.7
...
Required-by: kedro
```

5. kedro --help

```python
Traceback (most recent call last):
  File "/home/fgaudin/.pyenv/versions/test/bin/kedro", line 8, in <module>
    sys.exit(main())
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/kedro/framework/cli/cli.py", line 264, in main
    cli_collection = KedroCLI(project_path=Path.cwd())
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/kedro/framework/cli/cli.py", line 182, in __init__
    self._metadata = bootstrap_project(project_path)
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/kedro/framework/startup.py", line 184, in bootstrap_project
    configure_project(metadata.package_name)
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/kedro/framework/project/__init__.py", line 219, in configure_project
    settings.configure(settings_module)
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/dynaconf/base.py", line 182, in configure
    self._wrapped = Settings(settings_module=settings_module, **kwargs)
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/dynaconf/base.py", line 235, in __init__
    self.validators.validate(
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/dynaconf/validator.py", line 417, in validate
    validator.validate(self.settings, only=only, exclude=exclude)
  File "/home/fgaudin/.pyenv/versions/3.8.11/envs/test/lib/python3.8/site-packages/dynaconf/validator.py", line 197, in validate
    self._validate_items(
TypeError: _validate_items() got an unexpected keyword argument 'only'
```

## Expected Result
No crash :wink: 

## Problem
The problem comes from the file `kedro.framework.project.__init__.py` and this class:

```python
class _IsSubclassValidator(Validator):
    """A validator to check if the supplied setting value is a subclass of the default class"""

    def _validate_items(self, settings, env=None):
        super()._validate_items(settings, env)

        default_class = self.default(settings, self)
        for name in self.names:
            setting_value = getattr(settings, name)
            if not issubclass(setting_value, default_class):
                raise ValidationError(
                    f"Invalid value `{setting_value.__module__}.{setting_value.__qualname__}` "
                    f"received for setting `{name}`. It must be a subclass of "
                    f"`{default_class.__module__}.{default_class.__qualname__}`."
                )

```

In `dynaconf`, the signature changed: https://github.com/rochacbruno/dynaconf/blob/master/dynaconf/validator.py#L207-L213

```python
    def _validate_items(
        self,
        settings: Any,
        env: Optional[str] = None,
        only: Optional[Union[str, Sequence]] = None,
        exclude: Optional[Union[str, Sequence]] = None,
    ) -> None:
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): 0.17.4
* Python version used (`python -V`): 3.8.11
* Operating system and version: Ubuntu 20.04

**Comment by datajoely** [comment]
Hi @FlorianGD thanks for this - they released a breaking change between 3.1.5 and 3.1.6. Would you please pin your requirements to 3.1.5 whilst we work on a proper fix?

**Comment by FlorianGD** [comment]
Yes, that's what I did! Thank you

**Comment by deepyaman** [comment]
Fixed in #893
